src/nbt/tag.rs

```diff
@@ -25,11 +25,21 @@ pub enum NbtTag {
 }
 
 impl NbtTag {
-    pub fn id(&self) -> u8 {
+    /// Returns the numeric id associated with the data type.
+    pub const fn id(&self) -> u8 {
+        // See https://doc.rust-lang.org/reference/items/enumerations.html#pointer-casting
         unsafe { *(self as *const Self as *const u8) }
     }
 
-    pub fn serialize_raw(&self) -> Bytes {
+    pub fn serialize(&self) -> Bytes {
+        let mut bytes = BytesMut::new();
+        bytes.put_u8(self.id());
+        bytes.put(self.serialize_data());
+        bytes.freeze()
+    }
+
+    /// Serializes tag data into bytes.
+    pub fn serialize_data(&self) -> Bytes {
```

Comment:
Okay so you just said this in comment why comment when you CAN LITERALLY NAME THE FUNCTION